sql-plugin/src/main/scala/com/nvidia/spark/rapids/SchemaUtils.scala

```diff
@@ -329,10 +331,13 @@ object SchemaUtils {
       builder: NestedBuilder[T, V],
       schema: StructType,
       writeInt96: Boolean = false,
-      parquetFieldIdEnabled: Boolean = false): T = {
+      parquetFieldIdEnabled: Boolean = false,
+      nullable: Boolean = true): T = {
```

Comment:
Is true an appropriate default value?  I noticed that every occurrence of this had to either be updated to explicitly pass `false` or a computed value, which makes me wonder if `false` would be the better default.  `false` would also match the prior behavior, respecting each top-level field's nullable setting rather than smashing it.